pkg/framer/list_time_series.go

```diff
@@ -42,61 +42,56 @@ func (f *timeSeriesSummaryFields) fields() data.Fields {
 
 func newTimeSeriesSummaryFields(length int) *timeSeriesSummaryFields {
 	return &timeSeriesSummaryFields{
-		alias:         fields.AliasField(length),
-		assetId:           fields.AssetIdField(length),
-		dataType:          fields.DataTypeField(length),
-		dataTypeSpec:      fields.DataTypeSpecField(length),
-		propertyId:  fields.PropertyIdField(length),
-		timeSeriesArn:  fields.TimeSeriesArnField(length),
-		timeSeriesId:  fields.TimeSeriesIdField(length),
-		timeSeriesCreationDate: fields.TimeSeriesCreationDateField(length),
-		timeSeriesLastUpdateDate:   fields.TimeSeriesLastUpdateDateField(length),
+		alias:                    fields.AliasField(length),
+		assetId:                  fields.AssetIdField(length),
+		dataType:                 fields.DataTypeField(length),
+		dataTypeSpec:             fields.DataTypeSpecField(length),
+		propertyId:               fields.PropertyIdField(length),
+		timeSeriesArn:            fields.TimeSeriesArnField(length),
+		timeSeriesId:             fields.TimeSeriesIdField(length),
+		timeSeriesCreationDate:   fields.TimeSeriesCreationDateField(length),
+		timeSeriesLastUpdateDate: fields.TimeSeriesLastUpdateDateField(length),
 	}
 }
 
 func (t TimeSeries) Frames(_ context.Context, _ resource.ResourceProvider) (data.Frames, error) {
 
 	length := len(t.TimeSeriesSummaries)
-
-	timeSeriesSummaryFields := newTimeSeriesSummaryFields(length)
+	tsf := newTimeSeriesSummaryFields(length)
 
 	for i, timeSeries := range t.TimeSeriesSummaries {
-
-	
-		if (timeSeries.Alias != nil) {
-			timeSeriesSummaryFields.alias.Set(i, *timeSeries.Alias)
-		} 
-		if (timeSeries.AssetId != nil) {
-			timeSeriesSummaryFields.assetId.Set(i, *timeSeries.AssetId)
+		if timeSeries.Alias != nil {
+			tsf.alias.Set(i, *timeSeries.Alias)
 		}
-		if (timeSeries.DataType != nil) {
-			timeSeriesSummaryFields.dataType.Set(i, *timeSeries.DataType)
+		if timeSeries.AssetId != nil {
+			tsf.assetId.Set(i, *timeSeries.AssetId)
 		}
-		if (timeSeries.DataTypeSpec != nil) {
-			timeSeriesSummaryFields.dataTypeSpec.Set(i, *timeSeries.DataTypeSpec)
+		tsf.dataType.Set(i, timeSeries.DataType)
```

Comment:
I get the following error `error="interface conversion: interface {} is types.PropertyDataType, not string"` when I tried to run this query:

<img width="718" alt="list-time-series-query" src="https://github.com/user-attachments/assets/6dda90d0-2966-4df6-b560-e0c3397eeb75" />

Changing it to the following works.

```go
tsf.dataType.Set(i, string(timeSeries.DataType))
```